shared/src/hover/HoverOverlay.tsx

```diff
@@ -107,14 +107,16 @@ export class HoverOverlay<A extends string> extends React.PureComponent<HoverOve
 
         this.subscription.add(
             this.props.platformContext.settings.subscribe(s => {
-                this.setState({
-                    showBadges:
-                        s.final &&
-                        !isErrorLike(s.final) &&
-                        s.final.experimentalFeatures &&
-                        // Enabled if true or null
-                        s.final.experimentalFeatures.showBadgeAttachments !== false,
-                })
+                if (s.final && !isErrorLike(s.final)) {
+                    // Default to true if experimentalFeatures or showBadgeAttachments are not set
+                    this.setState({
+                        showBadges:
+                            !s.final.experimentalFeatures ||
+                            s.final.experimentalFeatures.showBadgeAttachments !== false,
+                    })
+                } else {
+                    this.setState({ showBadges: false })
+                }
```

Comment:
You can keep it in one condition if you use `?.`, because `undefined !== false` :)
```suggestion
                this.setState({
                    showBadges:
                        s.final &&
                        !isErrorLike(s.final) &&
                        // Enabled if true, null or undefined
                        s.final.experimentalFeatures?.showBadgeAttachments !== false,
                })
```

Reply:
I tried this, it actually doesn't type check for me. `showBadges` is a boolean and the last expression is inferred to be `bool | null`. Maybe it's just vscode being a doofus.

What I committed I _think_ is the clearest way to get the intention of this value across (which is subtle, hence it being broken for over two releases).